docs/developer_docs.md

```diff
@@ -0,0 +1,164 @@
+              order.
+   - returntype the sorted vector
+   
+#### Quick sort
+- int* quickSort(int* array, int low, int high, int &stepCounter, int mode = 1);
+  - Purpose of the function:
+    - it sorts the array's elements descending or ascending order based on the mode's value.
+   - parameteres:
+     - array: array with numbers,
+     - low: the first element of the array
+     - high: the last element of the array
+     - stepCounter: number of steps while the program does the sorting
+     - mode: an integer, if that integer is 0 it should sort the input numbers in descending order and if it is 1 it should sort the
+              input numbers in increasing order. If the second parameter is not provided it should sort the given list is increasing
+              order.
+   - returntype the sorted array
+   
+- std::vector<int> quickSort(std::vector<int> &vector, int low, int high, int &stepCounter, int mode = 1);
+  - Purpose of the function:
+    - it sorts the vector's elements descending or ascending order based on the mode's value.
+   - parameteres:
+     - vector: vector with numbers,
+     - low: the first element of the vector
+     - high: the last element of the vector
+     - stepCounter: number of steps while the program does the sorting
+     - mode: an integer, if that integer is 0 it should sort the input numbers in descending order and if it is 1 it should sort the
+              input numbers in increasing order. If the second parameter is not provided it should sort the given list is increasing
+              order.
+   - returntype the sorted vector
+   
+#### Bubble sort
+- std::vector<int> bubbleSort(std::vector<int> a, int &stepCounter, int n = 1);
+  - Purpose of the function:
+    - it sorts the vector's elements descending or ascending order based on the mode's value.
+   - parameteres:
+     - a: vector with numbers,
+     - stepCounter: number of steps while the program does the sorting
+     - n: an integer, if that integer is 0 it should sort the input numbers in descending order and if it is 1 it should sort the
+          input numbers in increasing order. If the second parameter is not provided it should sort the given list is increasing order.
+   - returntype the sorted vector
+   
+- int* bubbleSort(int arr[], int size, int &stepCounter, int n = 1);
+  - Purpose of the function:
+    - it sorts the array's elements descending or ascending order based on the mode's value.
+   - parameteres:
+     - arr[]: vector with numbers,
+     - stepCounter: number of steps while the program does the sorting
+     - n: an integer, if that integer is 0 it should sort the input numbers in descending order and if it is 1 it should sort the
+          input numbers in increasing order. If the second parameter is not provided it should sort the given list is increasing order.
+   - returntype the sorted array
+   
+ #### functions in main
```

Comment:
this part is a bit messy, remove it or make it just as good as the previous one :)